From a8f9dba38a0b66d72f8417439447b04c0e6ed629 Mon Sep 17 00:00:00 2001
From: jinghua <xigu@marvell.com>
Date: Sun, 18 Sep 2016 21:58:03 +0800
Subject: [PATCH 0511/1345] dts: a3700: reorganize dts files of devel-board

commit  05b71c1020cb347f55993630c4e5176e39ba281d from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Armada3700 rev1 devel-board used to support Z1
  chip. It will be reused for A0 chip. Thus, the
  rev1 board is now tied to A0 chip. Meanwhile,
  the dts(i) files of Z1 chips was removed.
- The rev2 devel-board is a upgrade version of
  rev1 devel-board. There is no changes from the
  IO's perspective. Both boards will use the same
  dts file.
- The patch also moved 'dma-coherent' property to
  'soc' node in armada-37xx.dtsi file.

Change-Id: Ieac60c3f5ea2c794fc8b338bb7ca7fb094773916
Signed-off-by: jinghua <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32834
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Wilson Ding <dingwei@marvell.com>
(cherry picked from commit df3e00f15539c10044d1fcdddcdd1b1d2acdff2f)
Reviewed-on: http://vgitil04.il.marvell.com:8080/32921
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/Makefile               |    3 -
 arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi    |   53 -----
 .../boot/dts/marvell/armada-3720-community.dts     |    2 +-
 .../boot/dts/marvell/armada-3720-rz-db-sfp.dts     |   73 ------
 .../boot/dts/marvell/armada-3720-rz-db-sgmii1.dts  |   65 ------
 arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts  |  238 --------------------
 arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi    |   63 ------
 arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi    |   67 ------
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi       |    1 +
 9 files changed, 2 insertions(+), 563 deletions(-)
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-3720-rz-db-sfp.dts
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-3720-rz-db-sgmii1.dts
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi
 delete mode 100644 arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi

diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index ae47b31..504338f 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -3,9 +3,6 @@ dtb-$(CONFIG_ARCH_BERLIN) += berlin4ct-dmp.dtb
 dtb-$(CONFIG_ARCH_BERLIN) += berlin4ct-stb.dtb
 
 # Mvebu SoC Family
-dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-rz-db.dtb
-dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-rz-db-sgmii1.dtb
-dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-rz-db-sfp.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-sgmii1.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3720-db-sfp.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi b/arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi
deleted file mode 100644
index 30a4e57..0000000
--- a/arch/arm64/boot/dts/marvell/armada-371x-rz.dtsi
+++ /dev/null
@@ -1,53 +0,0 @@
-/*
- * Device Tree Include file for Marvell Armada 371x family of RZ SoCs
- * (also named 88F3710-RZ)
- *
- * Copyright (C) 2016 Marvell
- *
- * Gregory CLEMENT <gregory.clement@free-electrons.com>
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPL or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This file is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This file is distributed in the hope that it will be useful
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-#include "armada-37xx-rz.dtsi"
-
-/ {
-	model = "Marvell Armada 3710 RZ SoC";
-	compatible = "marvell,armada3710", "marvell,armada3700";
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community.dts b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
index 31838b6..1e4be59 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-community.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
@@ -43,7 +43,7 @@
 
 /dts-v1/;
 
-#include "armada-372x-rz.dtsi"
+#include "armada-372x.dtsi"
 
 / {
 	model = "Marvell Armada 3720 Community Board";
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-rz-db-sfp.dts b/arch/arm64/boot/dts/marvell/armada-3720-rz-db-sfp.dts
deleted file mode 100644
index e46fe00..0000000
--- a/arch/arm64/boot/dts/marvell/armada-3720-rz-db-sfp.dts
+++ /dev/null
@@ -1,73 +0,0 @@
-/*
- * Device Tree file for Marvell Armada 3720 RZ development board
- * (DB-88F3720-DDR3-RZ)
- * Copyright (C) 2016 Marvell
- *
- * Ken Ma <make@marvell.com>
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPL or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This file is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This file is distributed in the hope that it will be useful
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-#include "armada-3720-rz-db.dts"
-/* This file enables eth1 to work in SGMII mode with 1g or 2.5g fixed-link,
- * eth0 still works in RGMII mode.
- */
-/ {
-	soc {
-		internal-regs {
-			eth1: ethernet@40000 {
-				phy-mode = "sgmii";
-				status = "okay";
-				fixed-link {
-					/* Here "speed" is set to 1000, GBE MAC is running in 1G mode, */
-					/* but the actual MAC speed may be 1 Gbps or 2.5 Gbps, */
-					/* it's up to the corresponding SERDES speed in u-boot fdt comphy node. */
-					/* If SERDES speed is set to 3.125G in u-boot, it implies sgmii 2.5 Gbps; */
-					/* if SERDES speed is set to 1.25G in u-boot, it implies sgmii 1 Gbps. */
-					speed = <1000>;
-					full-duplex;
-				};
-			};
-			usb3@58000 {
-				status = "disabled";
-			};
-		};
-	};
-};
-
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-rz-db-sgmii1.dts b/arch/arm64/boot/dts/marvell/armada-3720-rz-db-sgmii1.dts
deleted file mode 100644
index 644af1b..0000000
--- a/arch/arm64/boot/dts/marvell/armada-3720-rz-db-sgmii1.dts
+++ /dev/null
@@ -1,65 +0,0 @@
-/*
- * Device Tree file for Marvell Armada 3720 RZ development board
- * (DB-88F3720-DDR3-RZ)
- * Copyright (C) 2016 Marvell
- *
- * Ken Ma <make@marvell.com>
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPL or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This file is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This file is distributed in the hope that it will be useful
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-#include "armada-3720-rz-db.dts"
-/* This file enables eth1 to work on SERDES SGMII1 interface and
- * use SMI to do PHY polling, meanwhile eth0 still works on RGMII mode.
- */
-/ {
-	soc {
-		internal-regs {
-			eth1: ethernet@40000 {
-				phy-mode = "sgmii";
-				phy = <&phy1>;
-				status = "okay";
-			};
-			usb3@58000 {
-				status = "disabled";
-			};
-		};
-	};
-};
-
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts
deleted file mode 100644
index 4a9af0a..0000000
--- a/arch/arm64/boot/dts/marvell/armada-3720-rz-db.dts
+++ /dev/null
@@ -1,238 +0,0 @@
-/*
- * Device Tree file for Marvell Armada 3720 RZ development board
- * (DB-88F3720-DDR3-RZ)
- * Copyright (C) 2016 Marvell
- *
- * Gregory CLEMENT <gregory.clement@free-electrons.com>
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPL or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This file is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This file is distributed in the hope that it will be useful
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-/dts-v1/;
-
-#include "armada-372x-rz.dtsi"
-
-/ {
-	model = "Marvell Armada 3720 RZ Development Board DB-88F3720-DDR3";
-	compatible = "marvell,armada-3720-db", "marvell,armada3720", "marvell,armada3710";
-
-	chosen {
-		stdout-path = "serial0:115200n8";
-	};
-
-	memory {
-		device_type = "memory";
-		reg = <0x00000000 0x00000000 0x00000000 0x20000000>;
-	};
-
-	soc {
-		internal-regs {
-			spi0: spi@10600 {
-				status = "okay";
-
-				spi-flash@0 {
-					#address-cells = <1>;
-					#size-cells = <1>;
-					compatible = "jedec,spi-nor";
-					reg = <0>; /* Chip select 0 */
-					spi-max-frequency = <104000000>;
-
-					pinctrl-names = "default";
-					pinctrl-0 = <&spi0_pins &spi_quad_pins>;
-
-					partition@0 {
-						label = "Bootloader";
-						reg = <0 0x200000>;
-					};
-					partition@200000 {
-						label = "U-boot Env";
-						reg = <0x200000 0x10000>;
-					};
-					partition@210000 {
-						label = "Linux";
-						reg = <0x210000 0xDF0000>;
-					};
-				};
-			};
-
-			i2c0: i2c@11000 {
-				status = "okay";
-				expander0: pca9555@22 {
-					compatible = "nxp,pca9555";
-					reg = <0x22>;
-					gpio-controller;
-					#gpio-cells = <2>;
-				};
-			};
-
-			i2c1: i2c@11080 {
-				status = "okay";
-			};
-
-			/*
-			* Exported on the micro USB connector CON32
-			* through an FTDI
-			*/
-			uart0: serial@12000 {
-				status = "okay";
-			};
-
-			eth0: ethernet@30000 {
-				phy-mode = "rgmii-id";
-				phy = <&phy0>;
-				status = "okay";
-			};
-
-			mdio@32004 {
-				phy0: ethernet-phy@0 {
-					reg = <0>;
-				};
-				phy1: ethernet-phy@1 {
-					reg = <1>;
-				};
-			};
-
-			u3d@50000 {
-				status = "disabled";
-			};
-
-			udc@54100 {
-				status = "disabled";
-			};
-
-			/* CON31 */
-			usb3@58000 {
-				status = "okay";
-				usb-phy = <&usb3_phy>;
-			};
-
-			usb@5e000 {
-				status = "okay";
-			};
-
-			interrupt-controller@74840 {
-				status = "okay";
-			};
-
-			sdhci1: sdhci@d0000 {
-				wp-inverted;
-				bus-width = <4>;
-				cd-gpios = <&gpio_sb 2 GPIO_ACTIVE_LOW>;
-				vqmmc-supply = <&vcc_sd_reg1>;
-				status = "okay";
-
-				pinctrl-names = "default";
-				pinctrl-0 = <&sdio_pins &cd_pins>;
-			};
-
-			sdhci0: sdhci@d8000 {
-				xenon,fixed-1-8v-pad-ctrl;
-				non-removable;
-				bus-width = <8>;
-				tuning-count = <6>;
-				status = "okay";
-
-				pinctrl-names = "default";
-				pinctrl-0 = <&mmc_pins>;
-			};
-
-			/* CON3 */
-			sata@e0000 {
-				target-supply = <&reg_5v_sata>;
-				status = "okay";
-			};
-			vcc_sd_reg1: regulator@0 {
-				compatible = "regulator-gpio";
-				regulator-name = "vcc_sd1";
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-boot-on;
-
-				gpios = <&gpio_sb 22 GPIO_ACTIVE_HIGH>;
-				gpios-states = <0>;
-				states = <1800000 0x1
-					  3300000 0x0>;
-
-				enable-active-high;
-			};
-		};
-
-		reg_sata: pwr-sata {
-			compatible = "regulator-fixed";
-			regulator-name = "pwr_en_sata";
-			regulator-min-microvolt = <12000000>;
-			regulator-max-microvolt = <12000000>;
-			enable-active-high;
-			regulator-boot-on;
-			gpio = <&expander0 2 GPIO_ACTIVE_HIGH>;
-		};
-
-		reg_5v_sata: v5-sata {
-			compatible = "regulator-fixed";
-			regulator-name = "v5.0-sata";
-			regulator-min-microvolt = <5000000>;
-			regulator-max-microvolt = <5000000>;
-			vin-supply = <&reg_sata>;
-		};
-
-		reg_12v_sata: v12-sata {
-			compatible = "regulator-fixed";
-			regulator-name = "v12.0-sata";
-			regulator-min-microvolt = <12000000>;
-			regulator-max-microvolt = <12000000>;
-			vin-supply = <&reg_sata>;
-		};
-	};
-
-	usb3_phy: usb3-phy {
-		compatible = "usb-nop-xceiv";
-		vcc-supply = <&exp_usb3_vbus>;
-	};
-
-	exp_usb3_vbus: usb3-vbus {
-		compatible = "regulator-fixed";
-		regulator-name = "usb3-vbus";
-		regulator-min-microvolt = <5000000>;
-		regulator-max-microvolt = <5000000>;
-		enable-active-high;
-		regulator-always-on;
-		gpio = <&expander0 1 GPIO_ACTIVE_HIGH>;
-	};
-};
-
diff --git a/arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi b/arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi
deleted file mode 100644
index 0bef261..0000000
--- a/arch/arm64/boot/dts/marvell/armada-372x-rz.dtsi
+++ /dev/null
@@ -1,63 +0,0 @@
-/*
- * Device Tree Include file for Marvell Armada 372x family of RZ SoCs
- * (also named 88F3720-RZ)
- *
- * Copyright (C) 2016 Marvell
- *
- * Gregory CLEMENT <gregory.clement@free-electrons.com>
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPL or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This file is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This file is distributed in the hope that it will be useful
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-#include "armada-37xx-rz.dtsi"
-
-/ {
-	model = "Marvell Armada 3720 RZ SoC";
-	compatible = "marvell,armada3720", "marvell,armada3710";
-
-	cpus {
-		cpu@1 {
-			device_type = "cpu";
-			compatible = "arm,cortex-a53","arm,armv8";
-			reg = <0x1>;
-			enable-method = "psci";
-		};
-	};
-
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi
deleted file mode 100644
index 0273b7d..0000000
--- a/arch/arm64/boot/dts/marvell/armada-37xx-rz.dtsi
+++ /dev/null
@@ -1,67 +0,0 @@
-/*
- * Device Tree Include file for Marvell Armada 37xx family of RZ SoCs.
- *
- * Copyright (C) 2016 Marvell
- *
- * Gregory CLEMENT <gregory.clement@free-electrons.com>
- *
- * This file is dual-licensed: you can use it either under the terms
- * of the GPL or the X11 license, at your option. Note that this dual
- * licensing only applies to this file, and not this project as a
- * whole.
- *
- *  a) This file is free software; you can redistribute it and/or
- *     modify it under the terms of the GNU General Public License as
- *     published by the Free Software Foundation; either version 2 of the
- *     License, or (at your option) any later version.
- *
- *     This file is distributed in the hope that it will be useful
- *     but WITHOUT ANY WARRANTY; without even the implied warranty of
- *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *     GNU General Public License for more details.
- *
- * Or, alternatively
- *
- *  b) Permission is hereby granted, free of charge, to any person
- *     obtaining a copy of this software and associated documentation
- *     files (the "Software"), to deal in the Software without
- *     restriction, including without limitation the rights to use
- *     copy, modify, merge, publish, distribute, sublicense, and/or
- *     sell copies of the Software, and to permit persons to whom the
- *     Software is furnished to do so, subject to the following
- *     conditions:
- *
- *     The above copyright notice and this permission notice shall be
- *     included in all copies or substantial portions of the Software.
- *
- *     THE SOFTWARE IS PROVIDED , WITHOUT WARRANTY OF ANY KIND
- *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
- *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
- *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
- *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY
- *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
- *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
- *     OTHER DEALINGS IN THE SOFTWARE.
- */
-
-#include "armada-37xx.dtsi"
-
-/ {
-	model = "Marvell Armada 37xx RZ SoC";
-
-	soc {
-		internal-regs {
-			npinctrl: north-bridge-pinctrl@13830 {
-				compatible = "marvell,armada-3700-z1-nb-pinctrl";
-			};
-
-			spinctrl: south-bridge-pinctrl@18830 {
-				compatible = "marvell,armada-3700-z1-sb-pinctrl";
-			};
-
-			usb3@58000 {
-				interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
-			};
-		};
-	};
-};
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index ee24abf..4845429 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -93,6 +93,7 @@
 			compatible = "simple-bus";
 			/* 32M internal register @ 0xd000_0000 */
 			ranges = <0x0 0x0 0xd0000000 0x2000000>;
+			dma-coherent;
 
 			spi0: spi@10600 {
 				compatible = "marvell,armada-3700-spi";
-- 
1.7.9.5

